#94 Realease notes for v1.1.0 (#101)
Inbibo/shift_docs: docs/release_notes.md

# Release Notes

## Version 1.1.0

### Added

#### Core

- Workflow execution on a dedicated thread. This allows users to inspect and author different workflows while executing.
- `SHIFT_SINGLE_THREADED` environment variable to switch the threaded execution behaviour.
- Workflow execution progress bar in the Shift UI.
- Board styling for higlighting the execution status of the workflow.
- Ability to request the interruption of an executing workflow with the "Clear Execution" button. The button will now clear the workflow only if pressed while no execution is running on the current board.
- Ability to rename a node by double clicking on its name with the left mouse button.

#### Integration & Resources

> - Nuke: Menu file to setup Shift tools inside Nuke.
> - Nuke (Beta): "ShiftWorkflow" Nuke node to execute Shift workflow inside the Nuke node graph.

Author: What do you think about this?
```suggestion
**Nuke**
- Added a Menu file to set up Shift tools inside Nuke.
- (Beta) Included a new "ShiftWorkflow" Nuke node to execute Shift workflow inside the Nuke node graph. This node is in beta, meaning that its behaviour might change in the future.
```

> - Nuke (Beta): "ShiftWorkflow" Nuke node to execute Shift workflow inside the Nuke node graph.

Reviewer: Good

#### API

> - Added `shift.utils.dcc_threading` module containing utilities to handle threaded execution in DCC context
> - Added `shift.utils.decorators` module with `mainThread` decorator for SOperator.execute methods to force execution on main thred in Maya and Nuke

Reviewer 2: Remember to add the full stop at the end of the sentence to be aligned with the rest.
```suggestion
- Added `shift.utils.dcc_threading` module containing utilities to handle threaded execution in DCC context.
- Added `shift.utils.decorators` module with `mainThread` decorator for SOperator.execute methods to force execution on main thred in Maya and Nuke.
```

- Added `disableEditing` method to `ParameterWidget` class to enable/disable editing capabilities of the Inspector widgets.

### Improvements

#### Core

> - Set Focus Execution to `False` by default. The global execution status is now tracked on the progress bar. *Shift-1218*

Author: I thought it should have been for imporvements as well. Ok!

Committing the suggestion

#### Catalogs

**Standard Catalog (Version 1.1.0)**

> - Added `mainThread` boolean plug to PythonScript node to force execution on main thread in DCC context. This is needed to use DCC Python API calls for Maya or Nuke in the node's script. *Shift-1218*

Author: This logic is specific for DCC context for Maya and Nuke. I would not say of an application and I would avoid the "for example" as those are the only cases in which it has effect.

> - Added `mainThread` boolean plug to PythonScript node to force execution on main thread in DCC context. This is needed to use DCC Python API calls for Maya or Nuke in the node's script. *Shift-1218*

Reviewer: Ok, leave it as it is then

**CreativeSoftware Catalog (Version 1.1.0)**

> - This catalog's nodes have been made compatible with threaded execution feature in Maya. *Shift-1218*

Author: What do you think about:

```suggestion
- This catalog's nodes have been made compatible with the new threaded execution feature in Maya and Nuke. *Shift-1218*
```

All node's logic has been wrapped so it does not make sense to say that only the ones for Maya and Nuke have been modified.

> - This catalog's nodes have been made compatible with threaded execution feature in Maya. *Shift-1218*

Reviewer: Is just that the way it is worded sounds that the threaded execution is a change in Maya and Nuke

> - This catalog's nodes have been made compatible with threaded execution feature in Maya. *Shift-1218*

Author: What about

```suggestion
- This catalog's nodes have been made compatible with threaded execution feature. *Shift-1218*
```

> - This catalog's nodes have been made compatible with threaded execution feature in Maya. *Shift-1218*

Reviewer: This is good

# Release Notes

## Version 1.1.0

### Added

#### Core

- Shift workflows now run on a dedicated thread, enabling users to inspect and author other workflows and interact with the UI while there is an ongoing execution.

> - An environment variable, `SHIFT_SINGLE_THREADED`, has been added to enable or disable threaded execution.

Reviewer 2: Question, here we're mentioning the use of the environment variable, should we also mention the option in the menu to change it on-the-fly?

Author: yes, let me do it.

Author: done
